src/estimagic/parameters/consolidate_constraints.py

```diff
@@ -644,10 +645,14 @@ def _check_consolidated_weights(weights, param_names):
     relevant_names = [param_names[i] for i in weights.columns]
 
     if n_constraints > n_params:
-        raise ValueError(msg_too_many + msg_general.format(relevant_names, weights))
+        raise InvalidConstraintError(
+            msg_too_many + msg_general.format(relevant_names, weights)
+        )
 
     if np.linalg.matrix_rank(weights) < n_constraints:
-        raise ValueError(msg_rank + msg_general.format(relevant_names, weights))
+        raise InvalidConstraintError(
```

Comment:
This line is not covered by any tests. Could you create a simple test case to get this error? If its too much work you can merge.